feat(cli): make `Spinner.message` able to be changed on-the-fly

timreichen opened the pull request:
- makes `Spinner.message` property public

Reason: Allows to change message during active spinner.

Example:
```ts
import { Spinner } from "https://deno.land/std@$STD_VERSION/cli/spinner.ts";
import { delay } from "https://deno.land/std@$STD_VERSION/async/delay.ts";

const spinner = new Spinner( { message: "Loading..." });
spinner.start();
await delay(1000);
spinner.message = "Almost done..."
await delay(1000);
spinner.stop();
```

iuioiua reviewed (approved):
I like the idea of this. Thank you.

It does make me think that this same change could be applied to some of the other properties.

timreichen commented:
> I like the idea of this. Thank you.
> 
> It does make me think that this same change could be applied to some of the other properties.

Like `color` and `interval`?

iuioiua commented:
Yep 👍🏾

timreichen commented:
> Yep 👍🏾

Should that be part of this PR or a separate one?

iuioiua commented:
Let's have it part of this one.

timreichen commented:
Actually, `interval` might need special treatment, else a change won't do anything.

timreichen commented:
Ok, I updated the code so it works when changing `interval` and `color`.
While at it, I removed the obsolete `#active` prop and made some adjustments for better readability.

iuioiua commented on cli/spinner.ts:
```diff
@@ -63,11 +63,11 @@ export interface SpinnerOptions {
   #color?: Color;
   #intervalId: number | undefined;
-  #active = false;
+  #frameIndex = 0;
```
What's the purpose of this change?

iuioiua reviewed (changes requested):
Frankly, I don't get some of these changes. I didn't think making color and friends would introduce such logic.

Sorry to change my mind, but can we revert to focusing on making `Spinner.message` public? Sorry for my previous suggestion. We should always keep PRs focused.

timreichen commented on cli/spinner.ts:
```diff
@@ -43,17 +43,17 @@ export interface SpinnerOptions {
    */
   interval?: number;
+  /** The message to display next to the spinner. */
+  message?: string;
```
I put internal props `#spinner` and `#interval` on top and changeable props at after.

timreichen commented:
> Frankly, I don't get some of these changes. I didn't think making color and friends would introduce such logic.
> 
> Sorry to change my mind, but can we revert to focusing on making `Spinner.message` public? Sorry for my previous suggestion. We should always keep PRs focused.

Ok, I reverted the changes and will open other PRs for the `color` and `interval`. There are some catches to be aware of so it works as intended, so the changes are not just to remove the hash.

timreichen commented on cli/spinner.ts:
```diff
@@ -79,11 +79,51 @@ export class Spinner {
+  }
+  get color() {
+    return this.#color;
+  }
```
because the actual value is determined by `COLORS[value]`

iuioiua reviewed (approved):
Nice! LGTM. Thank you.
